qcodes/tests/parameter/test_parameter_scale_offset.py

```diff
@@ -69,11 +69,15 @@ def test_scale_and_offset_raw_value_iterable(values, offsets, scales):
     # test that scale and offset does not change the default behaviour
     p(values)
     assert p.raw_value == values
+    old_value = p()
 
     # test setting scale and offset does not change anything
     p.scale = scales
     p.offset = offsets
     assert p.raw_value == values
+    new_value = p()
+
+    assert type(new_value) == type(old_value)
```

Comment:
do we have an explicit test like this for numpy-array-valued parameter?

Reply:
I do not understand your question. Are you suggesting this test condition should be a test in itself?

Reply:
> Are you suggesting this test condition should be a test in itself?

nope. I am asking whether there exists a test case (could be that it's already part of the parametrization of this test) where we assert scale/offset working on a parameter that returns numpy arrays (as opposed to scalars, tuples, lists, etc)

Reply:
there is a condition tested for when all set values, offset, scale and parameter returned values are a numpy array, and this test checks for the correct data result. But the return data types itself are not checked. This assert condition which I added checks explicitly for that.

Reply:
> values, offset, scale and parameter returned values are a numpy array

is this really true? the way i read the hypothesis strategy, only python lists are tested for, not numpy arrays. perhaps i'm misreading :)

Reply:
hmm, you are right... I need to understand how hypothesis strategy is working better. What I was referring to in my previous reply is the test on line 87 .. that's just a test for data.